src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,841 @@ static DamageTable D_80AD2550 = {
+            break;
+
+        case 0x1FD6:
+            this->unk_1F4 = 0x1FD7;
+            break;
+
+        case 0x1FD8:
+            this->unk_1F4 = 0x1FD9;
+            break;
+
+        case 0x1FDA:
+            this->unk_1F4 = 0x1FDB;
+            break;
+
+        case 0x1FDC:
+            this->unk_1F4 = 0x1FDD;
+            break;
+
+        case 0x1FDE:
+            this->unk_1F4 = 0x1FDF;
+            break;
+
+        case 0x1FE0:
+            this->unk_1F4 = 0x1FE1;
+            break;
+
+        case 0x1FE2:
+            this->unk_1F4 = 0x1FE3;
+            break;
+
+        case 0x1FE4:
+            this->unk_1F4 = 0x1FE5;
+            break;
+
+        case 0x1FE6:
+            this->unk_1F4 = 0x1FE7;
+            break;
+
+        case 0x1FE8:
+            this->unk_1F4 = 0x1FE9;
+            break;
+
+        case 0x1FEA:
+            this->unk_1F4 = 0x1FEB;
+            break;
+
+        case 0x1FEC:
+            this->unk_1F4 = 0x1FED;
+            break;
+
+        case 0x1FEE:
+            this->unk_1F4 = 0x1FEF;
+            break;
+
+        case 0x1FF0:
+            this->unk_1F4 = 0x1FF1;
+            break;
+
+        case 0x1FF2:
+            this->unk_1F4 = 0x1FF3;
+            break;
+
+        case 0x1FF4:
+            this->unk_1F4 = 0x1FF5;
+            break;
+
+        case 0x1FF6:
+            this->unk_1F4 = 0x1FF7;
+            break;
+
+        case 0x1FF8:
+            this->unk_1F4 = 0x1FF9;
+            break;
+
+        case 0x1FFA:
+            this->unk_1F4 = 0x1FFB;
+            break;
+
+        case 0x1FFC:
+            this->unk_1F4 = 0x1FFD;
+            break;
+
+        case 0x1FD7:
+        case 0x1FDB:
+        case 0x1FDD:
+        case 0x1FDF:
+        case 0x1FE1:
+        case 0x1FE7:
+        case 0x1FF1:
+        case 0x1FF3:
+        case 0x1FF7:
+        case 0x1FF9:
+
+        case 0x1FFB:
+            this->unk_1F4 = 0x1FD3;
+            break;
+
+        case 0x1FD9:
+        case 0x1FE3:
+        case 0x1FE9:
+        case 0x1FEB:
+        case 0x1FED:
+        case 0x1FEF:
+        case 0x1FF5:
+
+        case 0x1FFD:
+            this->unk_1F4 = 0x1FD4;
+            break;
+
+        case 0x1FE5:
+            this->unk_1F4 = 0x1FFE;
+            break;
+
+        case 0x1FD3:
+        case 0x1FD4:
+
+        case 0x1FFE:
+            this->unk_1F4 = 0x1FD5;
+            this->unk_1EA |= 0x20;
+    }
+
+    func_801518B0(globalCtx, this->unk_1F4, &this->actor);
+}
+
+void func_80AD1398(EnOsn* this) {
+    this->cutscene = this->actor.cutscene;
+    if (((gSaveContext.inventory.items[SLOT_OCARINA] == ITEM_NONE) ||
+         (INV_CONTENT(ITEM_MASK_DEKU) == ITEM_MASK_DEKU)) &&
+        ((this->cutscene = ActorCutscene_GetAdditionalCutscene(this->cutscene),
+          (gSaveContext.inventory.items[SLOT_OCARINA] != ITEM_NONE)) ||
+         (INV_CONTENT(ITEM_MASK_DEKU) == ITEM_MASK_DEKU))) {
+        this->cutscene = ActorCutscene_GetAdditionalCutscene(this->cutscene);
+    }
+}
```

Comment:
```suggestion
void func_80AD1398(EnOsn* this) {
    this->cutscene = this->actor.cutscene;
    if ((gSaveContext.inventory.items[SLOT_OCARINA] == ITEM_NONE) ||
         (INV_CONTENT(ITEM_MASK_DEKU) == ITEM_MASK_DEKU)) {
         this->cutscene = ActorCutscene_GetAdditionalCutscene(this->cutscene)
         if ((gSaveContext.inventory.items[SLOT_OCARINA] != ITEM_NONE) ||
         (INV_CONTENT(ITEM_MASK_DEKU) == ITEM_MASK_DEKU)) {
            this->cutscene = ActorCutscene_GetAdditionalCutscene(this->cutscene);
        }
    }
}
```
I think it would be much cleaner breaking up this if-statement into two parts so that you don't have this awkward `this->cutscene` assignment within the conditional. I would double check my brackets as it's hard to edit on GitHub, but this should be broken up (assuming it still matches)